subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -221,9 +221,20 @@ static bool direct_method_process(struct topic_parser_data *topic,
 	}
 
 	/* Get request ID */
-	evt.data.method.rid = atoi(topic->prop_bag[0].value);
+	for (size_t i = 0; i < topic->prop_bag_count; i++) {
```

Comment:
If I'm not mistaken this block of code is equal to the one added on line: 272. This could be compressed into one function call.

Reply:
Yeah, thought about that, but left it like this for now as they store the property bag value in different members in the `evt.data` union